include/cantera/thermo/CoverageDependentSurfPhase.h

```diff
@@ -0,0 +1,617 @@
+    std::string name_j; //!< name of species which thermodynamics are dependent on [integer]
+    vector_fp enthalpy_params; /*!< array of three parameters to calculate coverage-dependent enthalpy:
+                                *   slope of enthalpy change in the first region [J/kmol], slope of
+                                *   enthalpy change in the second region [J/kmol], and coverage
+                                *   dividing first and second region [dimensionless]
+                                */
+    vector_fp entropy_params; /*!< array of three parameters to calculate coverage-dependent entropy:
+                               *   slope of entropy change in the first region [J/kmol/K], slope of
+                               *   entropy change in the second region [J/kmol/K], and coverage
+                               *   dividing first and second region [dimensionless]
+                               */
+};
+
+//! Set of parameters modifying SurfPhase enthalpy and entropy based on fractional surface coverages 
+//! using a interpolative model.
+struct InterpolativeDependency
+{
+    //! Constructor
+    //! @param name_k_  name of species which thermodynamics are calculated for [integer]
+    //! @param name_j_  name of species which thermodynamics are dependent on [integer]
+    /*! @param enthalpy_coverages_  array of coverages for coverage-dependent 
+     *                              enthalpy interpolation [dimensionless]
+     */
+    /*! @param enthalpies_ array of enthalpies at corresponding coverages 
+     *                     in enthalpy-coverages [J/kmol]
+     */
+    /*! @param entropy_coverages_  array of coverages for coverage-dependent 
+     *                             entropy interpolation [dimensionless]
+     */
+    /*! @param entropies_  array of entropies at corresponding coverages 
+     *                     in entropy-coverages [J/kmol/K]
+     */
+    
+    InterpolativeDependency(std::string name_k_, std::string name_j_,
+                            vector_fp enthalpy_coverages_, vector_fp enthalpies_,
+                            vector_fp entropy_coverages_, vector_fp entropies_):
+                            name_k(name_k_), name_j(name_j_),
+                            enthalpy_coverages(enthalpy_coverages_),
+                            enthalpies(enthalpies_),
+                            entropy_coverages(entropy_coverages_),
+                            entropies(entropies_) {}
+    InterpolativeDependency() {}
+    std::string name_k; //!< name of species which thermodynamics are calculated for [integer]
+    std::string name_j; //!< name of species which thermodynamics are dependent on [integer]
+    vector_fp enthalpy_coverages; /*!< array of coverages for coverage-dependent 
+                                   *   enthalpy interpolation [dimensionless]
+                                   */
```

Comment:
If the description of a variable doesn't easily fit at the end of the line, I'd write it as:
```suggestion
    //! array of coverages for coverage-dependent enthalpy interpolation [dimensionless]
    vector_fp enthalpy_coverages;
```